spec/mempool/v0/quint/mempoolv0.qnt

```diff
@@ -0,0 +1,524 @@
+        Chain::addNewBlock(txs),
+    }
+
+    // This node is the proposer: create the next block with txs from the mempool.
+    action BlockExecutor_MempoolUpdate_proposer(node) = all {
+        require(node.mempoolTxs().nonEmpty()),
+        require(node.MempoolHeight() == Chain::latestHeight),
+        val nextBlockHeight = node.MempoolHeight() + 1
+        // Note that block may be empty.
+        nondet txs = oneOf(node.mempoolTxs().powerset())
+        BlockExecutor_MempoolUpdate_proposer_(node, txs, nextBlockHeight)
+    }
+
+    //--------------------------------------------------------------------------
+    // Mempool reactor (P2P) actions
+    //--------------------------------------------------------------------------
+    /* Receive a Txs message from a peer, and calls CheckTx on each Tx. */
+    // [Reactor.Receive]: https://github.com/CometBFT/cometbft/blob/111d252d75a4839341ff461d4e0cf152ca2cc13d/mempool/reactor.go#L93
+    action P2P_ReceiveTxs(node) = all {
+        node.setStep("P2P_ReceiveTxs", Map("node" -> node)),
+        require(node.Network::incomingMsgs().nonEmpty()),
+        nondet msg = oneOf(node.Network::incomingMsgs())
+        all {
+            node.CheckTx(msg.tx, msg.senderId),
+            node.Network::receiveMsg(msg),
+            History::unchanged,
+            Chain::unchanged,
+        }
+    }
+
+    /* The reactor loops through its mempool and sends the tx one by one to
+     * each of its peers. */
+    //-- __subaction__: this action is not part of the next-state relation
+    action P2P_SendTx_(node: NodeId, peer: NodeId): bool = all {
+        node.setStep("P2P_SendTx", Map("node" -> node, "peer" -> peer)),
+        require(not(node.mempoolIsEmpty())),
+        nondet tx = oneOf(node.mempoolTxs())
+        val msg = { tag: "Txs", senderId: node, tx: tx }
+        all {
+            // If the msg was not already sent to this peer.
+            require(peer.Network::incomingMsgs().forall(m => m != msg)),
+            // If the peer is not a tx's sender.
+            require(peer.notIn(node.sendersFor(tx))),
+            Network::sendTo(msg, peer),
+            node.setError(noError),
+            node.History::recordGossippedTx(tx),
+            mempool' = mempool,
+            cache' = cache,
+            mempoolHeight' = mempoolHeight,
+            ABCI::unchanged,
+            Chain::unchanged,
+        }
+    }
+
+    // [Reactor.broadcastTxRoutine]: https://github.com/CometBFT/cometbft/blob/5049f2cc6cf519554d6cd90bcca0abe39ce4c9df/mempool/reactor.go#L132
+    action P2P_SendTx(node: NodeId): bool =
+        nondet peer = oneOf(Network::Peers.get(node))
+        node.P2P_SendTx_(peer)
+
+    /* The reactor sends at once a tx in the mempool to all its peers. */
+    // [Reactor.broadcastTxRoutine]: https://github.com/CometBFT/cometbft/blob/5049f2cc6cf519554d6cd90bcca0abe39ce4c9df/mempool/reactor.go#L132
+    action P2P_BroadcastTx(node: NodeId): bool = all {
```

Comment:
Initially I had a `P2P_SendTx` action but it introduced a lot of additional, unnecessary steps to the trace, one for each peer. Since each peer process the received message when they need to, it doesn't make any real difference to put the message in the peer's inbox all at once.